src/NServiceBus.RavenDB.Tests/SubscriptionStorage/When_receiving_an_unsubscription_message.cs

```diff
@@ -25,4 +25,23 @@ public class When_receiving_an_unsubscribe_message : RavenDBPersistenceTestBase
 
         Assert.IsEmpty(clients);
     }
+
+    [Test]
+    public async Task All_subscription_entries_for_specified_message_types_should_be_removed_even_if_version_changes()
+    {
+        await SubscriptionIndex.CreateAsync(store);
+
+        var storage = new SubscriptionPersister(store);
+        var context = new ContextBag();
+
+        await storage.Subscribe(TestClients.ClientA, MessageTypes.MessageA, context);
```

Comment:
Should we also add here saving document with different version? probably needs to be created/saved manually.